src/main/kotlin/com/coder/gateway/CoderGatewayConnectionProvider.kt

```diff
@@ -93,31 +109,38 @@ class CoderGatewayConnectionProvider : GatewayConnectionProvider {
 
             indicator.text = "Configuring Coder CLI..."
-            cli.configSsh(client.agentNames(workspaces))
-
-            // TODO: Ask for these if missing.  Maybe we can reuse the second
-            //  step of the wizard?  Could also be nice if we automatically used
-            //  the last IDE.
-            if (parameters[IDE_PRODUCT_CODE].isNullOrBlank()) {
-                throw IllegalArgumentException("Query parameter \"$IDE_PRODUCT_CODE\" is missing")
-            }
-            if (parameters[IDE_BUILD_NUMBER].isNullOrBlank()) {
-                throw IllegalArgumentException("Query parameter \"$IDE_BUILD_NUMBER\" is missing")
-            }
-            if (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank()) {
-                throw IllegalArgumentException("One of \"$IDE_PATH_ON_HOST\" or \"$IDE_DOWNLOAD_LINK\" is required")
-            }
-
-            // Check that both the domain and the redirected domain are
-            // allowlisted.  If not, check with the user whether to proceed.
-            verifyDownloadLink(parameters)
-
-            // TODO: Ask for the project path if missing and validate the path.
-            val folder = parameters[FOLDER] ?: throw IllegalArgumentException("Query parameter \"$FOLDER\" is missing")
-
-            parameters
-                .withWorkspaceHostname(CoderCLIManager.getHostName(deploymentURL.toURL(), agent.name))
-                .withProjectPath(folder)
+            cli.configSsh(client.agentNames(workspaces).toSet().toList())
+
+
+            val openDialog = if (parameters[IDE_PRODUCT_CODE].isNullOrBlank())
+                true
+            else if (parameters[IDE_BUILD_NUMBER].isNullOrBlank())
+                true
+            else if (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank())
+                true
+            else if (parameters[FOLDER].isNullOrBlank())
+                true
+            else
+                false
+
+            val params = if (openDialog) {
+                val view = CoderWorkspaceStepView{}
+                ApplicationManager.getApplication().invokeAndWait {
+                    view.init(
+                        CoderWorkspacesStepSelection(agent, workspace, cli, client, workspaces)
+                    )
+                    val dialog = SelectWorkspaceIDEDialog(view.component)
+                    dialog.show()
+                }
+                val p = parameters.toMutableMap()
+                listOf(IDE_PRODUCT_CODE, IDE_BUILD_NUMBER, IDE_DOWNLOAD_LINK, IDE_PATH_ON_HOST, PROJECT_PATH).forEach { prop ->
+                    view.data()[prop]?.let { value -> p[prop] = value }
+                }
+                p
+            } else
+                parameters.withProjectPath(parameters[FOLDER]!!)
+
+            params.withWorkspaceHostname(CoderCLIManager.getHostName(deploymentURL.toURL(), "${workspace.name}.${agent.name}"))
```

Comment:
I am not sure how this ever worked ... this was just using the `agent.name` instead of `${workspace.name}.${agent.name}` so it never matched any ssh config rules.

Reply:
Good catch, I think I recently broke this because `agent` here used to be a `WorkspaceAgentListModel` and I have been refactoring to pass a regular `WorkspaceAgent` around, but I missed this name.